### DISCUSSION TITLE
Next.js redirect if there is no cookie

### DISCUSSION THREAD

**Discussion by domanskyi** [vercel/next.js#25653] [Help]
Is there a way to redirect the user from checkout to the join page ONLY if there is no cookie `user_token`. Next.js allows to set only strings and undefined as `value`, but I need to validate that there is no cookie:
```js
  redirects: async () => [
    {
      source: '/checkout',
      has: [
        {
          type: 'cookie',
          key: 'user_token',
          value: '',
        },
      ],
      permanent: true,
      destination: '/join',
    },
  ],
```

I tried to use regex for empty string, but it did not work:
```js
  redirects: async () => [
    {
      source: '/checkout',
      has: [
        {
          type: 'cookie',
          key: 'user_token',
          value: '(^$)',
        },
      ],
      permanent: true,
      destination: '/join',
    },
  ],
```

**Comment by KingMatrix1989** [comment]
I have the same question

**Comment by vikasg603** [comment]
Same question.

**Comment by snax4a** [comment]
I also would like to skip redirect if some header value is preset.
To achieve that a 'hasNot' field should be added to next.js redirects

**Comment by adamjakab** [comment]
Same question here. A "simple" NOT operator would be good to negate has items.

In the meanwhile I am using this (not very elegant but working) workaround in a specific page:
Checking for the existence of a specific cookie and if it is not there redirecting to the /auth page.
```
export const getServerSideProps: GetServerSideProps = async (ctx) => {
  const auth_cookie_name = process.env.NEXT_PUBLIC_APP_AUTH_COOKIE_NAME as string
  //console.log('Cookies:', ctx.req.cookies)

  if (!ctx.req.cookies.hasOwnProperty(auth_cookie_name)) {
    //console.log("NO AUTH COOKIE")
    ctx.res.statusCode = 307
    ctx.res.setHeader('Location', '/auth')
  }

  return {props: {}}
}
```

**Comment by BuddhaBing** [comment]
@ijjk can you help with this? It's been open for nearly a year with no valid answer...

**Comment by alexrequelme** [comment]
Now, with Next.js 13, you have the option to utilize the 'missing' field in the redirects configuration. You can find more info in the docs https://nextjs.org/docs/pages/api-reference/next-config-js/redirects. For example: 

```js
async redirects() {
    return [
      {
        source: "/home",
        missing: [
          {
            type: "cookie",
            key: "user_session",
          },
        ],
        permanent: false,
        destination: "/auth/signin",
      },
    ];
  },
```
